File: iPad/VerBank-CSTSv3-IOSClientApi/ClientStation/api/info/InfoOperator_TRADESERV1015.m
//
//  InfoOperator_TRADESERV1015.m
//  VerBank-IOSClientStation
//
//  Created by Allone on 15/8/31.
//  Copyright (c) 2015年 zhanglei. All rights reserved.
//

#import "InfoOperator_TRADESERV1015.h"

#import "Info_TRADESERV1015.h"
#import "API_IDEventCaptain.h"
#import "MTP4CommDataInterface.h"
#import "TradeApi.h"

@implementation InfoOperator_TRADESERV1015

- (void)onInfo:(InfoFather *)info {
    Info_TRADESERV1015 *tinfo = (Info_TRADESERV1015 *)info;
    [API_IDEventCaptain fireEventChanged:DATA_ON_PriceWarning_Reached eventData:[tinfo getPriceWarning]];
    if ([[tinfo getPriceWarning] getOrigin] == ROLETYPE_USER) {
        [NSThread detachNewThreadSelector:@selector(runReached:) toTarget:self withObject:[[tinfo getPriceWarning] getGuid]];
    }
}

- (void)runReached:(NSString *)guid {
    [[TradeApi getInstance] readPriceWarning:guid];
}

@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/client/station/api/info/InfoOperator_TRADESERV1015.java
package jedi.ex01.client.station.api.info;

import jedi.ex01.CSTS3.comm.info.InfoFather;
import jedi.ex01.CSTS3.comm.info.Info_TRADESERV1015;
import jedi.ex01.client.station.api.event.API_IDEventCaptain;
import jedi.ex01.client.station.api.event.API_IDEvent_NameInterface;

public class InfoOperator_TRADESERV1015 extends InfoOperator {

	@Override
	public void onInfo(InfoFather _info) {
		Info_TRADESERV1015 info = (Info_TRADESERV1015) _info;
		API_IDEventCaptain.fireEventChanged(API_IDEvent_NameInterface.DATA_ON_PriceWarning_Reached, info
				.getPriceWarning());

	}

}
